Simplenote/NoteEditorViewController+Swift.swift

```diff
@@ -941,16 +941,77 @@ extension NoteEditorViewController {
         SearchQuery(searchText: toolbarView.searchField.stringValue)
     }
 
-    private func updateKeywordsHighlight() {
+    private var highlightedRanges: [NSRange] {
         let keywords = searchQuery.keywords
 
         guard !noteEditor.isFirstResponder, !keywords.isEmpty else {
-            noteEditor.highlightedRanges = []
-            return
+            return []
         }
 
         let slice = noteEditor.attributedString().string.contentSlice(matching: keywords)
-        noteEditor.highlightedRanges = slice?.nsMatches ?? []
+        return slice?.nsMatches ?? []
+    }
+
+    private func updateKeywordsHighlight() {
+        let ranges = highlightedRanges
+
+        noteEditor.highlightedRanges = ranges
+
+        createSearchMapViewIfNeeded()
+        searchMapView?.update(with: searchBarPositions(with: ranges))
+    }
+}
+
+
+// MARK: - Search Map
+//
+extension NoteEditorViewController {
+    private func createSearchMapViewIfNeeded() {
+        guard searchMapView == nil else {
+            return
+        }
+
+        let searchMapView = SearchMapView()
+        searchMapView.translatesAutoresizingMaskIntoConstraints = false
+
+        view.addSubview(searchMapView)
+
+        NSLayoutConstraint.activate([
+            searchMapView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            searchMapView.widthAnchor.constraint(equalToConstant: EditorMetrics.searchMapWidth),
+            searchMapView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor),
+            searchMapView.topAnchor.constraint(equalTo: scrollView.topAnchor, constant: SplitItemMetrics.editorScrollerTopInset)
+        ])
+
+        self.searchMapView = searchMapView
+    }
+
+    /// Returns position relative to the total text container height.
+    /// Position value is from 0 to 1
+    ///
+    private func searchBarPositions(with searchRanges: [NSRange]) -> [CGFloat] {
```

Comment:
WDYT if `searchBarPositions` and `textContainerHeightForSearchMap` were moved over SPTextView?

We could rename this API as follows (and shrink a bit the VC?)

```
searchBarPositions` > `relativeLocationsForText(in ranges: [NSRange]) -> [CGFloat]
```

Reply:
Good idea, let me try :-)